HW_module_8.py

```diff
@@ -0,0 +1,151 @@
+        print(f'Your default directory is "{os.getcwd()}"')
+        is_true = True
+        while is_true:
+            answer = int(input(
+                f'If you want to ingest your file from default directory - enter 1, if you want to change it - enter 2: '))
+            if answer == 1:
+                while True:
+                    try:
+                        file_name = input('Enter the file name with its format: ')
+                        with open(file_name, "r", encoding='utf-8') as read_file:
+                            lict_of_dicts = json.load(read_file)
+                        print('Okay, such file exists')
+                        path_for_remove = str(file_name)
+                        is_true = False
+                        break
+                    except FileNotFoundError:
+                        print('No file with such name. Try again')
+                    except PermissionError:
+                        print('No file with such name. Try again')
+
+            elif answer == 2:
+                while True:
+                    try:
+                        change_path = input('Enter the file path: ')
+                        if not os.path.isdir(change_path):
+                            os.mkdir(change_path)
+                        os.chdir(change_path)
+                        print(f'Now you directory is "{os.getcwd()}"')
+                        break
+                    except SyntaxError:
+                        print('You made a mistake. Try again')
+                    except FileNotFoundError:
+                        print('You made a mistake. Try again')
+                while True:
+                    try:
+                        file_name = input('Enter the file name with its format: ')
+                        with open(file_name, "r", encoding='utf-8') as read_file:
+                            lict_of_dicts = json.load(read_file)
+                        print('Such file exists')
+                        path_for_remove = os.path.join(str(change_path), str(file_name))
+                        os.chdir(original_path)
+                        is_true = False
+                        break
+                    except FileNotFoundError:
+                        print('No file with such name. Try again')
+                    except PermissionError:
+                        print('No file with such name. Try again')
+            else:
+                print('Try again')
+        return lict_of_dicts, path_for_remove
+
+    def publishing(self):
+        lict_of_dicts, path_for_remove = self.read_file()
+
+        insert_into_file = []
+        try:
+            for i, d in enumerate(lict_of_dicts):
+                data = []
+                if d["type"] == 'News':
+                    if len(d["text"]) > 0 and len(d["location"]) > 0 and len(d["date"]) > 0:
+                        data.append('News ----------------------------'+'\n')
```

Comment:
Why not invoke here the News class here? All formatting is contained inside the news_message method.